packages/remix-react/components.tsx

```diff
@@ -322,20 +322,34 @@ export { NavLink };
  * @see https://remix.run/api/remix#link
  */
 let Link = React.forwardRef<HTMLAnchorElement, RemixLinkProps>(
-  ({ to, prefetch = "none", ...props }, forwardedRef) => {
+  ({ to: inputTo, prefetch = "none", children, ...props }, forwardedRef) => {
+    let to = inputTo.toString();
+    let isExternal = /^[a-z]+:/.test(to);
     let href = useHref(to);
+
     let [shouldPrefetch, prefetchHandlers] = usePrefetchBehavior(
       prefetch,
       props
     );
+
+    if (isExternal) {
+      return (
+        <>
+          <a ref={forwardedRef} href={to} {...props} {...prefetchHandlers}>
+            {children}
+          </a>
+          {shouldPrefetch ? (
+            <link key={href} rel="prefetch" as="document" href={to} />
```

Comment:
I think you don't need the `as="document"` here, I'm using it on my blog without that and it works https://github.com/sergiodxa/sergiodxa.com/blob/ca1fa0d9c7f8d92322b01f009951bbfdd54c6214/app/root.tsx#L213-L215

Also when I tried to prefetch links to external websites the browser didn't re-used the prefetched response and did a new request.

Reply:
looks like `rel=prerender` might be what we're actually after https://developer.mozilla.org/en-US/docs/Web/HTML/Link_types/prerender

Reply:
Is there any browser supporting prerender? I tried in Safari and Chromium (Edge) and it did nothing 🤔

Reply:
back working on this, looks like pre-render was removed 🫠 https://developer.chrome.com/blog/prerender-pages/#a-brief-history-of-prerender

Reply:
now it's a script tag apparently and idk how to feel lol 
```html
<script type="speculationrules">
{
  "prerender": [
    {"source": "list",
     "urls": ["/home", "/about"]}
  ],
  "prefetch": [
    {"source": "list",
     "urls": ["https://en.wikipedia.org/wiki/Hamster_racing"],
     "requires": ["anonymous-client-ip-when-cross-origin"]}
  ]
}
</script>
```

Reply:
additional update: pre-render does work still in chrome, confirmed using https://stackoverflow.com/a/64330790
![image](https://user-images.githubusercontent.com/11698668/214408912-41731795-b86a-4b8b-892c-554b998b6e87.jpeg)

Reply:
Do we really _need_ to try to optimize external links to apps we don't control, especially if it's potentially not straightforward?

Reply:
_probably_ not, but would've been nice if it was a straightforward api

Reply:
yeah definitely